common/sgx/verifier.c

```diff
@@ -443,96 +444,114 @@ static oe_result_t _fill_with_known_claims(
             valid_until,
             sizeof(*valid_until)));
 
-        // SGX quote PCESVN
-        OE_CHECK(oe_sgx_add_claim(
-            &claims[claims_index++],
-            OE_CLAIM_SGX_PCE_SVN,
-            sizeof(OE_CLAIM_SGX_PCE_SVN),
-            &sgx_quote->pce_svn,
-            sizeof(uint16_t)));
-
         // quote QEID
+        uint8_t ueid[OE_UEID_SIZE] = {0};
+        ueid[0] = OE_UEID_TYPE_RAND;
+        memcpy(&ueid[1], sgx_quote->user_data, SGX_USERDATA_SIZE);
```

Comment:
Need to use the vendor_id field in the sgx_quote_header_t section of the SGX quote. This value needs to be extracted from the SGX quote, which can be implemented in  function oe_verify_quote_with_sgx_endorsements().

Also noticed that in common/sgx/verifier.c lines 454-460, claim OE_CLAIM_SGX_QE_ID is assigned with sgx_quote->user_data. Should it be assigned with the above vendor_id field instead?

Reply:
Hi @shnwc 
The UEID (universal Entity ID) here is being used to identify the quoting enclave on that particular platform/hardware.
The IETF draft draft says UEID may represent an entity/submodule within a hardware/platform - 
`It may identify the entire entity or a submodule.`
https://datatracker.ietf.org/doc/html/draft-ietf-rats-eat#section-3.4

sgx_quote->user_data provides us with exactly that data as described in pg 21 of https://download.01.org/intel-sgx/dcap-1.0.1/docs/Intel_SGX_ECDSA_QuoteGenReference_DCAP_API_Linux_1.0.1.pdf

`User Data - 20 Byte Array - 
Custom user-defined data. For the Intel® SGX
DCAP library, the first 16 bytes contain a QE
identifier that is used to link a PCK Cert to an
Enc(PPID). This identifier is consistent for
every quote generated with this QE on this
platform.`

Reply:
@RRathna thank you for your explanation.